atomic_reactor/plugins/pre_inject_yum_repo.py

```diff
@@ -9,75 +9,123 @@
+        self._ca_bundle_pem = None
+        self._dockerfile = None
+
+    def _inject_into_repo_files(self):
+        """Inject repo files into a relative directory inside the build context"""
+        host_repos_path = os.path.join(self.workflow.builder.df_dir, RELATIVE_REPOS_PATH)
+        self.log.info("creating directory for yum repos: %s", host_repos_path)
+        os.mkdir(host_repos_path)
+
+        for repo_filename, repo_content in self.workflow.files.items():
+            # Update every repo accordingly in a repofile
+            # input_buf ---- updated ----> updated_buf
+            with StringIO(repo_content) as input_buf, StringIO() as updated_buf:
+                for line in input_buf:
+                    updated_buf.write(line)
+                    # Apply sslcacert to every repo in a repofile
+                    if line.lstrip().startswith('[') and self._builder_ca_bundle:
+                        updated_buf.write(f'sslcacert=/tmp/{self._ca_bundle_pem}\n')
+
+                yum_repo = YumRepo(repourl=repo_filename,
+                                   content=updated_buf.getvalue(),
+                                   dst_repos_dir=host_repos_path,
+                                   add_hash=False)
+                yum_repo.write_content()
+
+    def _inject_into_dockerfile(self):
+        self._dockerfile.add_lines(
+            "ADD %s* %s" % (RELATIVE_REPOS_PATH, YUM_REPOS_DIR),
+            all_stages=True, at_start=True, skip_scratch=True
+        )
+
+        if self._builder_ca_bundle:
+            shutil.copyfile(
+                self._builder_ca_bundle,
+                os.path.join(self.workflow.builder.df_dir, self._ca_bundle_pem)
+            )
+            self._dockerfile.add_lines(
+                f'ADD {self._ca_bundle_pem} /tmp/{self._ca_bundle_pem}',
+                all_stages=True, at_start=True, skip_scratch=True
+            )
+
+        if not self.workflow.builder.dockerfile_images.base_from_scratch:
+            self._dockerfile.add_lines(*self._cleanup_lines())
+
     def run(self):
         """
         run the plugin
         """
         yum_repos = {k: v for k, v in self.workflow.files.items() if k.startswith(YUM_REPOS_DIR)}
         if not yum_repos:
             return
-        # absolute path in containers -> relative path within context
-        host_repos_path = os.path.join(self.workflow.builder.df_dir, RELATIVE_REPOS_PATH)
-        self.log.info("creating directory for yum repos: %s", host_repos_path)
-        os.mkdir(host_repos_path)
 
-        for repo, repo_content in self.workflow.files.items():
-            yum_repo = YumRepo(repourl=repo, content=repo_content, dst_repos_dir=host_repos_path,
-                               add_hash=False)
-            yum_repo.write_content()
+        self._dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        if self._dockerfile.baseimage is None:
+            raise RuntimeError("No FROM line in Dockerfile")
```

Comment:
Is this even valid Dockerfile?

Reply:
Empty Dockerfile should not be reasonable for an image build. The plugin raises this error originally.

Reply:
Ok